### PR TITLE
Cuda cores

### PR DESCRIPTION
**Pull request by QuietSwami**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,13 @@
 FROM python:3-alpine
 
+RUN apk add --no-cache docker=19.03.12-r0
```
Comment (cjdcordeiro) [inline Dockerfile]: why this package?

File: code/Dockerfile.gpu [hunk 2]
```diff
@@ -4,4 +4,5 @@ RUN apt update && apt install -y python3
 
 COPY check-cuda.py /
 
+CMD ["python3", "check-cuda.py"]
```
Comment (cjdcordeiro) [inline code/Dockerfile.gpu:7]: can you run this on an alpine image?

File: code/discovery.py [hunk 3]
```diff
@@ -162,29 +288,62 @@ def readRuntimeFiles(path):
...
+
         # GPU is present and able to be used
 
         if dockerVersion():
             logging.info('--gpus is available in Docker...')
 
         else:
             logging.info('--gpus is not available in Docker, but GPU usage is available')
+        
+        runtime['additional-assets'] = {'device-information': cudaCoresInformation(runtime['devices'], True)}
```
Comment (cjdcordeiro) [inline code/discovery.py]: did you try this against the testbed? does the `device-information` attr in Nuvla?

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]

File: code/Dockerfile.gpu (hunk 2 above)
Comment (QuietSwami) [reply code/Dockerfile.gpu:7]: No, there are issues with libs and drivers when running in Alpine.
Cuda is not supported in Alpine

File: Dockerfile (hunk 1 above)
Comment (QuietSwami) [reply Dockerfile]: Legacy code. I will remove as its not needed

Comment (cjdcordeiro) [reply Dockerfile]: ok. then please resolve when removed

File: code/discovery.py (hunk 3 above)
Comment (QuietSwami) [reply code/discovery.py]: Updated to reflect changes to Nuvla

**Pull request merged**